website/docs/r/service_dynamic_snippet_content_v1.html.markdown

```diff
@@ -0,0 +1,108 @@
+---
+layout: "fastly"
+page_title: "Fastly: service_dynamic_snippet_content_v1"
+sidebar_current: "docs-fastly-resource-service-v1"
+description: |-
+  Provides a means to define blocks of VCL logic that is inserted into your service through Fastly dynamic snippets.
+   
+---
+
+ # fastly_service_dynamic_snippet_content_v1
+
+ Defines content that represents blocks of VCL logic that is inserted into your service.  This resource will populate the content of a dynamic snippet and allow it to be manged without the creation of a new service verison. 
+ 
+ ~> **Warning:** Terraform will take precedence over any changes you make through the API. Such changes are likely to be reversed if you run terraform again.  
+
+ If Terraform is being used to populate the initial content of a dynamic snippet which you intend to manage via the API, then the lifecycle `ignore_changes` field can be used with the resource.  An example of this configuration is provided below.    
+
+
+ ## Example Usage
+
+ Basic usage:
+
+ ```hcl 
+ resource "fastly_service_v1" "myservice" {
+   name = "snippet_test1"
+ 
+   domain {
+     name    = "snippet.fastlytestdomain.com"
+     comment = "snippet test"
+   }
+ 
+   backend {
+     address = "tftesting.tftesting.net.s3-website-us-west-2.amazonaws.com"
+     name    = "AWS S3 hosting"
+     port    = 80
+   }
+ 
+   dynamicsnippet {
+     name     = "My Dynamic Snippet"
+     type     = "recv"
+     priority = 110
+   }
+ 
+   default_host = "tftesting.tftesting.net.s3-website-us-west-2.amazonaws.com"
+ 
+   force_destroy = true
+ }
+ 
+ resource "fastly_service_dynamic_snippet_content_v1" "my_dyn_content" {
+ 
+   service_id = fastly_service_v1.myservice.id
+   snippet_id = { for s in fastly_service_v1.myservice.dynamicsnippet : s.name => s.snippet_id }["My Dynamic Snippet"]
+ 
+   content = "if ( req.url ) {\n set req.http.my-snippet-test-header4 = \"true\";\n}"
+ 
+ }
+ ```
```

Comment:
It might be good to add a second`dynamicsnippet` to the example code so that it is clear for less experienced users to have a `"fastly_service_dynamic_snippet_content_v1"` resource for each `dynamicsnippet` block. @phamann what are your thoughts on this?

Reply:
This is a good point and I don't think it would hurt.

Reply:
Agreed 👍

Reply:
Applied the extra example with two dynamic snippet content resources.